### PR TITLE
fix: enable retry concurrency limit for readFile()

### LINKED ISSUE #18977: Bug: EMFILE: too many open files using Flat Config with very large projects
### Environment

Node version: v20.17.0
npm version: v10.8.2
Local ESLint version: v9.1.1 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22631

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>
We have a typescript project with 20000 ts/tsx files

### What did you expect to happen?

I expected ESLint to not encounter any errors.

### What actually happened?

Gettitng this error

Oops! Something went wrong! :(

ESLint: 9.1.1

Error: EMFILE: too many open files, open 'E:/**/*.tsx'
    at async open (node:internal/fs/promises:639:25)
    at async Object.readFile (node:internal/fs/promises:1242:14)
    at async Promise.all (index 8422)
    at async ESLint.lintFiles (E:\*\node_modules\eslint\lib\eslint\eslint.js:866:25)
    at async Object.execute (E:\*\node_modules\eslint\lib\cli.js:500:23)
    at async main (E:\S\*\node_modules\eslint\bin\eslint.js:153:22)

Oops! Something went wrong! :(

ESLint: 9.1.1

RangeError: Maximum call stack size exceeded
    at clearTimeout (node:timers:183:22)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:220:9)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)
    at #processQueue (E:\*\*\node_modules\@humanwhocodes\retry\dist\retrier.cjs:232:31)

### Link to Minimal Reproducible Example

[eslint_toomany_files_open.zip](https://github.com/user-attachments/files/17242328/eslint_toomany_files_open.zip)

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

In that repro follow the instructions in the Readme.md

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Updated `@humanwhocodes/retry` and set concurrency to 100.

fixes #18977

#### Is there anything you'd like reviewers to focus on?

I didn't add any new tests for this because I couldn't come up with one that failed without it. Functionally, this should result in no difference to the end user and assuming our existing tests pass, I think we're okay.

### CODE REVIEW

File: lib/eslint/eslint.js [hunk 1]
```diff
@@ -746,7 +746,7 @@ class ESLint {
         });
         const controller = new AbortController();
         const retryCodes = new Set(["ENFILE", "EMFILE"]);
-        const retrier = new Retrier(error => retryCodes.has(error.code));
+        const retrier = new Retrier(error => retryCodes.has(error.code), { signal: controller.signal, concurrency: 100 });
```
Comment (mdjermanovic) [inline lib/eslint/eslint.js:749]: It doesn't seem like the constructor supports option `signal`?

https://github.com/humanwhocodes/retry/blob/90317d3f133d45583296ef740fb8f8e2997c3903/src/retrier.js#L244

Comment (nzakas) [reply lib/eslint/eslint.js:749]: Doh! Put in the wrong place. Thanks.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

If possible, would like @fasttime to verify before today's release.

**Review by fasttime — APPROVED** [review]
LGMT, thanks!

**Pull request merged**